R/teal_slice.R

```diff
@@ -266,60 +274,59 @@ c.teal_slice <- function(...) {
   ans
 }
 
+#' @keywords internal
+#' @export
+as.list.teal_slice <- function(x, ...) {
```

Comment:
I think this doesn't make sense now to register S3 anymore. It was nice when we had `as.list.teal_slices` which recursively called `as.list.teal_slice`. I think you can change it to `slice_to_list` and avoid mentioning this in Namespace.

I started to dig in rstudio/shiny repo and I found this. https://github.com/rstudio/shiny/pull/79 There is `as.list` but has been deprecated in 2013.

Reply:
I disagree. `as.list` is exactly what this function does. Considering that we recently removed actual clout from the NAMESPACE I don't think adding one more method is a bad thing. We have methods for `c`, `[` in there, after all.

Reply:
Yeah, I am on @chlebowa front. as.list.teal_slice is a nice way to convert our object to a list. No idea what it does have in common with https://github.com/rstudio/shiny/pull/79 because we do use `reactiveValuesToList` but we also rearrange order of parameters and we assume that the input object is of teal_slice class and not any list. So this does not work for any list/object but rather for teal_slice and it creates an output that is a list that you can assume would be a regular output from as.list

Reply:
Fair enough, let's continue then 👍